src/gather_classes.rs

```diff
@@ -0,0 +1,732 @@
+
+use crate::trace_comparison::{DIFFERENT, SAME};
+
+use std::{
+    cmp::Ordering,
+    collections::{HashMap, HashSet},
+    io::{Read, Write},
+    path::{Display, Path, PathBuf},
+    str::FromStr,
+};
+
+use anyhow::{anyhow, Context};
+use itertools::Itertools;
+use starknet::{
+    contract,
+    core::types::{FieldElement, TransactionTraceWithHash},
+    macros::felt_hex,
+};
+use std::fs::OpenOptions;
+
+use crate::{
+    graph::{self, DependencyMap},
+    transaction_simulator::BlockSimulationReport,
+};
+
+use serde_json::{
+    json,
+    map::{Entry, Iter},
+    Value,
+};
+
+type ContractHash = FieldElement;
+type Selector = FieldElement;
+
+#[derive(Hash, Eq, PartialEq, Ord, PartialOrd, Clone)]
+struct EntryPoint {
+    contract_hash: ContractHash,
+    selector: Selector,
+}
+impl EntryPoint {
+    fn to_string(&self) -> String {
+        format!(
+            "({}, {})",
+            self.contract_hash.to_string(),
+            self.selector.to_string()
+        )
+    }
+
+    fn entry_point_from_str(entry_point_str: &str) -> Result<Self, anyhow::Error> {
+        let re = regex::Regex::new(r"\((?<contract_hash>.*?), (?<selector>.*?)\)")?;
+        match re.captures(entry_point_str) {
+            None => Err(anyhow!(
+                "Unexpected format for string: `{}`",
+                entry_point_str
+            )),
+            Some(caps) => Ok(Self {
+                contract_hash: FieldElement::from_str(&caps["contract_hash"])?,
+                selector: FieldElement::from_str(&caps["selector"])?,
+            }),
+        }
+    }
+}
+
+impl Serialize for EntryPoint {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.to_string().as_str())
+    }
+}
+impl<'de> Deserialize<'de> for EntryPoint {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        deserializer.deserialize_string(EntryPointVisitor {})
+    }
+}
+
+struct EntryPointVisitor;
+
+impl<'de> serde::de::Visitor<'de> for EntryPointVisitor {
+    type Value = EntryPoint;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("EntryPoint")
+    }
+
+    fn visit_borrowed_str<E>(self, v: &'de str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        self.visit_string(v.to_string())
+    }
+
+    fn visit_string<E>(self, v: String) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Self::Value::entry_point_from_str(v.as_str())
+            .map_err(|err| serde::de::Error::custom(format!("{err:?}")))
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Self::Value::entry_point_from_str(v)
+            .map_err(|err| serde::de::Error::custom(format!("{err:?}")))
+    }
+}
```

Comment:
I don't understand the **need** to implement this. Would you explain?

Reply:
No longer necessary since changing EntryPoint to a single value.